### ISSUE TITLE
[no-shadow] False positive: export enum

### ISSUE THREAD

**Issue by theneverstill** [typescript-eslint/typescript-eslint#2570]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

eslint config
```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```
(with and without the `args: 'none'` option)

foo.vue
```TS
import type { Route } from "vue-router";

...

const foo: Promise<Route> = this.$router.push({...});
```

tsconfig.json
```JSON
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "baseUrl": "./",
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "importHelpers": true,
    "lib": ["DOM", "DOM.Iterable", "ESNext", "ScriptHost"],
    "module": "ESNext",
    "moduleResolution": "node",
    "outDir": "./dist",
    "paths": {
      "@/*": ["./src/*"]
    },
    "resolveJsonModule": true,
    "strict": true,
    "target": "ESNext",
    "types": ["jest", "node"]
  },
  "include": [
    "bin/**/*.ts",
    "src/**/*.ts",
    "src/**/*.vue"
  ]
}
```

**Expected Result**

Similar to https://github.com/typescript-eslint/typescript-eslint/issues/2453, the type is being used and therefore it should not be flagged as unused.

**Actual Result**

```
error    'Route' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
```

**Additional Info**

This doesn't happen in `"@typescript-eslint/eslint-plugin": "^3.10.1",`

**Versions**

`"@typescript-eslint/eslint-plugin": "^4.1.1",`

**Comment by bradzacher** [comment]
I'm unable to repro this against master.
Likely you have old version of our tooling installed.

Could you please check the versions you have installed using one of the following?

```bash
$ npm list @typescript-eslint/eslint-plugin @typescript-eslint/parser
$ yarn list @typescript-eslint/eslint-plugin @typescript-eslint/parser
```

**Comment by theneverstill** [comment]
Thanks @bradzacher for the prompt and accurate response. npm neglected to inform me of the outdated peer dependency.

Nothing to see here, carry on. 👋

**Comment by theneverstill** [comment]
@bradzacher Let's update the title on this issue, but.. fixed one issue and broke another. Perhaps you can replicate this one:


```TS
export enum SignOutType {
  global = "GLOBAL",
  local = "LOCAL",
}
```

throws:
```
/src/enum/sign-out-type.ts
  1:13  error  'SignOutType' is already declared in the upper scope  no-shadow
```

Same version in initial description and previously it was working.

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

**Comment by theneverstill** [comment]
🤦 Sorry. I'm done wasting your time. Carry on.

**Comment by KostkaBrukowa** [comment]
Hey! I've had the same problem today and this issue saved me probably a lot of time. Thanks
